pkg/apis/common/v1/common.go

```diff
@@ -14,9 +14,27 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-// ReconcilerStatus represents status information about desired/available nodes.
-type ReconcilerStatus struct {
+type DeploymentHealth string
+
+const (
+	GreenHealth DeploymentHealth = "green"
+	RedHealth   DeploymentHealth = "red"
+)
+
+// DeploymentStatus represents status information about a deployment.
+type DeploymentStatus struct {
+	// AvailableNodes is the number of available instances.
```

Comment:
Not a strong opinion on this but I find it a bit suspicious and confusing that we are mixing `nodes` and `instances` here. Note that a deployment officially uses the term `replicas`. Wouldn't it be more accurate?

Reply:
👍 
I changed the comment to `// AvailableNodes is the number of replicas in the deployment.`
I think we want to keep `AvailableNodes` for backward-compatibility reasons (and consistency with Elasticsearch).